feat: deleteHeadBranch built-in documentation (#84)
reviewpad/docs: docs/guides/built-ins/actions/deleteHeadBranch/index.mdx

import CodeBlock from "@theme/CodeBlock";
import Example from "!!raw-loader!./example.yml";

**Description**:

Deletes the head branch of the pull request.

> This only works if the branch is stored in the same repository as the pull request target, i.e., if the pull request comes from the same repository and not from a fork.

Reviewer: This sentence can be a warning saying `This builtin does not work on pull request from forks.` WDYT?

Author: It seems right.

> The branch is deleted even if another pull request depends on the head branch. Therefore, GitHub will close the dependent pull requests.

Reviewer: Can you please clarify this `Therefore, GitHub will close the dependent pull requests.`?

If I have PR1 from A -> B and PR2 from C -> A. What happens to PR2 if I merge PR1 and delete the head branch?

Author: @ferreiratiago The PR2 is closed.

:::info INFO

> The action will only happen if and when the pull request is closed or merged.

Reviewer: > 
```suggestion
The action is only executed if the pull request is either closed or merged.
```

import CodeBlock from "@theme/CodeBlock";
import Example from "!!raw-loader!./example.yml";

**Description**:

Deletes the head branch of the pull request.

:::warning WARNING

This built-in does not work on pull request from forks.

> Deleting a branch will cause all pull requests that have the deleted branch as head or base to also be closed.

Reviewer: ```suggestion
Deleting a branch will cause all pull requests that have the deleted branch as head or base to be closed.
```

> The branch is deleted even if another pull request depends on the head branch. Therefore, GitHub will close the dependent pull requests.

Reviewer: I would suggest a warning for this.

```suggestion
:::warning WARNING

Deleting a branch will cause all pull requests that have the deleted branch as head to also be closed.

:::
```

Author: This warning is not correct because pull requests that have the deleted branch as head OR base will be closed.

Author: Also should I put both these warnings https://github.com/reviewpad/docs/pull/84#discussion_r999311426 together in one warning message?

Reviewer: > This warning is not correct because pull requests that have the deleted branch as head OR base will be closed.

Ok. Then we can update the info to be correct :)

> Also should I put both these warnings [#84 (comment)](https://github.com/reviewpad/docs/pull/84#discussion_r999311426) together in one warning message?

I would try to do that before placing them in different warning messages. The important thing is to make it clear :)

:::info INFO

> The action is only executed if the pull request is either closed or merged.

Reviewer: ```suggestion
This built-in is only executed if the pull request is either closed or merged.
```
